lib/dap/proto/ldap.rb

```diff
@@ -25,8 +49,15 @@ def self.decode_elem_length(data)
       len_bytes = length - 128
       return unless data.length > len_bytes + 2
 
-      if len_bytes == 2
+      if len_bytes == 1
+        length = data.byteslice(2, len_bytes).unpack('C')[0]
+      elsif len_bytes == 2
         length = data.byteslice(2, len_bytes).unpack('S>')[0]
+      elsif len_bytes == 3
+        a, b, c = data.byteslice(2, len_bytes).unpack('CCC')
+        a = a << 16
+        b = b << 8
+        length = a + b + c
```

Comment:
This entire if/else block could likely be simplified into one block that should handle all lengths, using the shifting techniques above, looping, and adding up the parts of the length chunks the appropriate number of times based on the value of `len_bytes`